journal/src/main/java/io/camunda/zeebe/journal/file/SegmentedJournal.java

```diff
@@ -211,10 +211,13 @@ public void flush() {
   @Override
   public JournalReader openReader() {
     final var stamped = acquireReadlock();
-    final var reader = new SegmentedJournalReader(this);
-    readers.add(reader);
-    releaseReadlock(stamped);
-    return reader;
+    try {
+      final var reader = new SegmentedJournalReader(this);
+      readers.add(reader);
+      return reader;
+    } finally {
+      releaseReadlock(stamped);
+    }
```

Comment:
Can we check whether there are more cases like this? I feel again uncomfortable with this lock strategy. Maybe just create an issue to invest time in this.

Reply:
I am not completely sure what you mean by that and what makes you uncomfortable.

To get a first overview of where locks are used, I scanned the source code for the usage of different locks (like `StampedLock` or `ReentrantLock`). For example, the `StampedLock` is used only in the `SegmentedJournal`, or the `ReentrantLock` is used in the class `CompletableActorFuture`. But I could not find any other usage of those locks or others in the source code.

As I am not into the entire code base yet, do you see/know any other locks being used? Also, what would be the separate issue about? Would it be about to look for other locking cases to ensure that the locks are properly released? Or would it be to investigate other locking options to avoid that way of locking? If it is the latter, could you please give a bit of context on what kind of problems you see with this locking strategy (or what makes you uncomfortable with those)?

Reply:
> As I am not into the entire code base yet, do you see/know any other locks being used? Also, what would be the separate issue about? Would it be about to look for other locking cases to ensure that the locks are properly released? Or would it be to investigate other locking options to avoid that way of locking? If it is the latter, could you please give a bit of context on what kind of problems you see with this locking strategy (or what makes you uncomfortable with those)?

For me it would be more like checking whether we have in other places the same issue

Reply:
Here is the ticket: #8382